core/src/mindustry/core/UI.java

```diff
@@ -316,7 +316,10 @@ public void showTextInput(String titleText, String dtext, int textLength, String
                 });
                 keyDown(KeyCode.ESCAPE, this::hide);
                 keyDown(KeyCode.BACK, this::hide);
-            }}.show();
+                show();
```

Comment:
show() call had to be moved inside instead of daisychaining, because the TextField would not be available at this point to set the focus on.